### PR TITLE
Allow changing of all config trough ENV

### PR DESCRIPTION
**Pull request by thijsvanloef**
## Context 

Prevent people from having to change files in order to change game options

## Choices

Add environment variables for every option in the game

## Test instructions

Set custom options and see if the PalWorldSettings.ini gets changed

## Checklist before requesting a review

* [x] I have performed a self-review of my code
* [x] I've added documentation about this change to the README.
* [x] I've not introduced breaking changes.

### CODE REVIEW

**Comment by thijsvanloef** [comment]
Still need to document everything

File: scripts/start.sh [hunk 1]
```diff
@@ -63,6 +63,222 @@ if [ ! "$(grep -s '[^[:space:]]' /palworld/Pal/Saved/Config/LinuxServer/PalWorld
...
+    sed -E -i "s/BuildObjectDamageRate=[+-]?([0-9]*[.])?[0-9]+/BuildObjectDamageRate=$BUILD_OBJECT_DAMAGE_RATE/" /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
+fi
+if [ -n "${BUILD_OBJECT_DETERIORATION_DAMAGE_RATE}" ]; then
+    echo "BUILD_OBJECT_DETERIORATION_DAMAGE_RATE=$BUILD_OBJECT_DETERIORATION_DAMAGE_RATE"
+    sed -E -i "s/BuildObjectDeteriorationDamageRate=[+-]?([0-9]*[.])?[0-9]+/BuildObjectDeteriorationDamageRate=$BUILD_OBJECT_DETERIORATION_DAMAGE_RATE/" /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
+fi
+if [ -n "${BUILD_OBJECT_DETERIORATION_DAMAGE_RATE}" ]; then
+    echo "BUILD_OBJECT_DETERIORATION_DAMAGE_RATE=$BUILD_OBJECT_DETERIORATION_DAMAGE_RATE"
+    sed -E -i "s/PalAutoHpRegeneRateInSleep=[+-]?([0-9]*[.])?[0-9]+/PalAutoHpRegeneRateInSleep=$BUILD_OBJECT_DETERIORATION_DAMAGE_RATE/" /palworld/Pal/Saved/Config/LinuxServer/PalWorldSettings.ini
+fi
```
Comment (MSpreckels) [inline scripts/start.sh:149]: duplicate

142-145

**Review by MSpreckels — COMMENTED** [review]
Remove configmap of palworldsetting.ini else it will get overridden

File: k8s/configmap.yaml [hunk 2]
```diff
@@ -15,79 +15,3 @@ data:
...
   SERVER_DESCRIPTION: ""
----
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  name: palworld-settings
-data:
-  PalWorldSettings.ini: >-
-    [/Script/OnlineSubsystemUtils.IpNetDriver]
-    NetServerMaxTickRate=60
```
Comment (MSpreckels) [inline k8s/configmap.yaml:26]: IMO this didn't work in the first place. We should make a PR and put this in Engine.ini.

**Review by MSpreckels — CHANGES_REQUESTED** [review]
You should also remove the configmap reference from deployment

https://github.com/thijsvanloef/palworld-server-docker/blob/5b72e8d926fbfc4c785ef9076b74c02e659e1d56/k8s/deployment.yaml#L42

https://github.com/thijsvanloef/palworld-server-docker/blob/5b72e8d926fbfc4c785ef9076b74c02e659e1d56/k8s/deployment.yaml#L49

**Comment by thijsvanloef** [comment]
@MSpreckels I think I got everything

**Comment by MSpreckels** [comment]
Now we have some parameters that appended to the startcommand and some that are replaced in the file. Wouldn't i make sense to even it out and set everything possible in file?

File: README.md [hunk 3]
```diff
@@ -198,6 +198,71 @@ The server will run a save before the backup if rcon is enabled.
 
 ## Editing Server Settings
 
+### With Environment Variables
+
+> [!IMPORTANT]
+>
+> These Environment Variables/Settings are subject to change since the game is still in beta
+
+| Variable                                  | Description                                                    | Default Value                                                                                | Allowed Value       PLAYER_STOMACH_DECREASE_RATE |
```
Comment (MSpreckels) [inline README.md:207]: how did PLAYER_STOMACH_DECREASE_RATE get up there in the header? :D

File: README.md [hunk 4]
```diff
@@ -198,6 +198,71 @@ The server will run a save before the backup if rcon is enabled.
...
+|-------------------------------------------|----------------------------------------------------------------|----------------------------------------------------------------------------------------------|--------------------------------------------------|
+| NETSERVERMAXTICKRATE                      | Changes the TickRate of the server.                            | 120                                                                                          | 30-120                                           |
```
Comment (MSpreckels) [inline README.md:209]: Not used right now

**Comment by thijsvanloef** [comment]
@MSpreckels I've only changed some appended variables to the PalWorldSettings.ini. Mainly ENV's that include texts, as spacing could cause some trouble with the ini file

**Comment by Paddy32810** [comment]
I change the ENV parameters through the stack editor, then I deploy

I want to add : 

BUILD_OBJECT_DETERIORATION_DAMAGE_RATE=0.000000
DEATH_PENALTY=Item
PAL_EGG_DEFAULT_HATCHING_TIME=0.00100
image


at the end, but it doesn't work. My eggs still take 36hours to hatch. How do I solve this ? Obviously changing Palwordsettings.ini doesn't work at all.

**Pull request merged**